components/brave_shields/browser/domain_block_navigation_throttle.cc

```diff
@@ -0,0 +1,165 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_shields/browser/domain_block_navigation_throttle.h"
+
+#include <memory>
+#include <utility>
+
+#include "base/bind.h"
+#include "base/task/post_task.h"
+#include "base/threading/thread_task_runner_handle.h"
+#include "brave/components/brave_shields/browser/ad_block_custom_filters_service.h"
+#include "brave/components/brave_shields/browser/ad_block_service.h"
+#include "brave/components/brave_shields/browser/domain_block_controller_client.h"
+#include "brave/components/brave_shields/browser/domain_block_page.h"
+#include "brave/components/brave_shields/browser/domain_block_tab_storage.h"
+#include "components/prefs/pref_service.h"
+#include "components/security_interstitials/content/security_interstitial_tab_helper.h"
+#include "components/user_prefs/user_prefs.h"
+#include "content/public/browser/browser_context.h"
+#include "content/public/browser/browser_thread.h"
+#include "content/public/browser/navigation_handle.h"
+#include "content/public/browser/web_contents.h"
+#include "content/public/browser/web_contents_user_data.h"
+#include "net/base/net_errors.h"
+
+namespace {
+
+bool ShouldBlockDomainOnTaskRunner(
```

Comment:
we shouldn't merge before we understand how it affects performance. Ideally, this check shouldn't be slower than SafeBrowsing, especially on android.

The easies measurement can be done using `SCOPED_UMA_HISTOGRAM_TIMER` or `UMA_HISTOGRAM_TIMES` or a similar function (results can be later observed on `brave://histograms`). Would be good to check the delay from `PostTask` to `OnShouldBlockDomain` and also time spent in this function

Reply:
added in https://github.com/brave/brave-core/commit/1d30188eb955ce9355f7426f5e603563abccf567